rllib/offline/estimators/direct_method.py

```diff
@@ -1,80 +1,22 @@
 from ray.rllib.policy import Policy
-from ray.rllib.policy.sample_batch import SampleBatch
+from ray.rllib.policy.sample_batch import MultiAgentBatch, DEFAULT_POLICY_ID
 from ray.rllib.utils.annotations import DeveloperAPI, override
 from ray.rllib.utils.framework import try_import_torch
-from ray.rllib.utils.numpy import convert_to_numpy
 from ray.rllib.utils.typing import SampleBatchType
-from ray.rllib.offline.estimators.fqe_torch_model import FQETorchModel
-from ray.rllib.offline.estimators.qreg_torch_model import QRegTorchModel
-from gym.spaces import Discrete
+from ray.rllib.utils.numpy import convert_to_numpy
 import numpy as np
 
 torch, nn = try_import_torch()
 
 logger = logging.getLogger()
 
 
-@DeveloperAPI
-def train_test_split(
-    batch: SampleBatchType,
-    train_test_split_val: float = 0.0,
-    k: int = 0,
-) -> Generator[Tuple[List[SampleBatch]], None, None]:
-    """Utility function to split a batch into training and evaluation episodes.
-
-    This function returns an iterator with either a train/test split or
-    a k-fold cross validation generator over episodes from the given batch.
-    By default, `k` is set to 0.0, which sets eval_batch = batch
-    and train_batch to an empty SampleBatch.
-
-    Args:
-        batch: A SampleBatch of episodes to split
-        train_test_split_val: Split the batch into a training batch with
-        `train_test_split_val * n_episodes` episodes and an evaluation batch
-        with `(1 - train_test_split_val) * n_episodes` episodes. If not
-        specified, use `k` for k-fold cross validation instead.
-        k: k-fold cross validation for training model and evaluating OPE.
-
-    Returns:
-        A tuple with two SampleBatches (eval_batch, train_batch)
-    """
-    if not train_test_split_val and not k:
-        logger.log(
-            "`train_test_split_val` and `k` are both 0;" "not generating training batch"
-        )
-        yield [batch], [SampleBatch()]
-        return
-    episodes = batch.split_by_episode()
-    n_episodes = len(episodes)
-    # Train-test split
-    if train_test_split_val:
-        train_episodes = episodes[: int(n_episodes * train_test_split_val)]
-        eval_episodes = episodes[int(n_episodes * train_test_split_val) :]
-        yield eval_episodes, train_episodes
-        return
-    # k-fold cv
-    assert n_episodes >= k, f"Not enough eval episodes in batch for {k}-fold cv!"
-    n_fold = n_episodes // k
-    for i in range(k):
-        train_episodes = episodes[: i * n_fold] + episodes[(i + 1) * n_fold :]
-        if i != k - 1:
-            eval_episodes = episodes[i * n_fold : (i + 1) * n_fold]
-        else:
-            # Append remaining episodes onto the last eval_episodes
-            eval_episodes = episodes[i * n_fold :]
-        yield eval_episodes, train_episodes
-    return
-
-
 @DeveloperAPI
 class DirectMethod(OffPolicyEstimator):
-    """The Direct Method estimator.
+    """The Direct Method estimator with a trainable Q-model.
```

Comment:
**todo**: More elaboration is needed on what a trainable Q-model means.

Reply:
Look at `FeatureImportance` for instance on how much detailed should the documentation be. An example config would also be nice.

Reply:
**todo**: I mean you need to explain more to a newbie what a direct method is. There should be a summary that is self contained without forcing the user to read the paper.